Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -195,18 +195,22 @@ static T GenerateVertexShader(API_TYPE api_type)
 		out.Write("{\n");
 		out.Write("coord = float4(0.0, 0.0, 1.0, 1.0);\n");
 		uid_data->texMtxInfo[i].sourcerow = xfmem.texMtxInfo[i].sourcerow;
+		uid_data->texMtxInfo[i].inputform = xfmem.texMtxInfo[i].inputform;
 		switch (texinfo.sourcerow)
 		{
 		case XF_SRCGEOM_INROW:
-			// The following assert was triggered in Super Smash Bros. Project M 3.6.
-			//_assert_(texinfo.inputform == XF_TEXINPUT_ABC1);
-			out.Write("coord = rawpos;\n"); // pos.w is 1
+			if (texinfo.inputform == XF_TEXINPUT_ABC1)
+				out.Write("coord.xyz = rawpos.xyz;\n");
+			else // XF_TEXINPUT_AB11
+				out.Write("coord.xy = rawpos.xy;\n");
```

Comment:
May you always just use the .xyz way, and flush .z afterwards to 1.0 again if texinfo.inputform != XF_TEXINPUT_ABC1?